File: src/main/java/social/alone/server/push/FcmToken.java
package social.alone.server.push;

import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;
import social.alone.server.user.User;

import javax.persistence.*;
import java.util.Optional;

@Entity
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
class FcmToken {

    @Id
    @GeneratedValue
    private Long id;

    @Column
    private String value;

    @ManyToOne
    private User user;

    FcmToken(String value, Optional<User> userMaybe) {
        this.value = value;
        this.user = userMaybe.orElse(null);
    }

}

File: src/main/kotlin/social/alone/server/push/FcmToken.kt
package social.alone.server.push

import org.hibernate.annotations.CreationTimestamp
import org.hibernate.annotations.UpdateTimestamp
import social.alone.server.user.User
import java.time.LocalDateTime
import javax.persistence.*

@Entity
@Table(name = "fcm_token", uniqueConstraints = [UniqueConstraint(columnNames = ["value"])])
class FcmToken(
        @Column val value: String,
        @ManyToOne var user: User? = null
) {

    @Id
    @GeneratedValue
    val id: Long? = null

    @CreationTimestamp
    private val createdAt: LocalDateTime = LocalDateTime.now()

    @UpdateTimestamp
    private var updatedAt: LocalDateTime = LocalDateTime.now()


    fun updateUser(user: User?){
        this.user = user;
    }

}
